### DISCUSSION TITLE
Add support for Ring Mailbox Sensor (2024)

### DISCUSSION THREAD

**Discussion by szhu25** [tsightler/ring-mqtt#925] [Feature Requests]
Hi!

Ring recently released a new Mailbox sensor (Gen 1, 2024) to replace all previous models. I grabbed one and setup it in my Ring account, but it does not show up on Ring-MQTT (addon). Following discussion #304, I have tried to use the [Ring client api](https://github.com/dgreif/ring) to dump the full discovery data, but that cli does not seem to be able to find the new sensor either. 
Is there anything I could do to help adding support for that sensor to Ring-MQTT?

Amazon Listing: https://www.amazon.com/Ring-Mailbox-Sensor-Years-Battery/dp/B0D6162SQC
Ring Listing: https://ring.com/products/ring-mailbox-sensor

**Comment by tsightler** [comment]
As is shown in both listings, this sensor does not connect with Ring hub devices, rather it is a standalone device that connects via Amazon Sidewalk network.  The device appears in the Ring app because they have implemented integration with Amazon Sidewalk directly in the app, but it is an entirely different protocol and outside of the scope of ring-client-api, and thus ring-mqtt, at this time.

**Reply by szhu25** [reply]
That makes sense. When I setup the device through Ring app, it does provide me with an option to connect through any Ring bridges (which I do not have at this moment) instead of Amazon Sidewalk. If I purchase a Hub bridge and connect the sensor using Hub, could it be discovered by ring-client-api? I will try to grab one when it's on sale if that could be possible.

**Reply by tsightler** [reply]
Maybe, I don't have any way to know for sure.  Certainly the old mailbox sensors did work this way, however, the challenge with the old sensors was always range since the device would need a direct Z-wave connection to the hub.  However, it's also possible that Ring is just using these other hub devices as Sidewalk bridges, in which case it wouldn't help at all.

**Reply by szhu25** [reply]
Just tested. Using a Ring Bridge instead of Amazon Sidewalk works and the device would be discovered by addon (but not the Ring integration). 
However, it is quite PITA because I had to remove the mailbox sensor from Ring account, find the setup QR code and scan it again... Thankfully I do not have to remove the battery cover from sensor (and the mailbox)

**Reply by tsightler** [reply]
Thanks for coming back to report.  Nice to know that there is still an option available even though it won't work in all cases.  I wish it was possible to do something with Sidewalk, but it feels like it would be a significant reverse engineering effort.  I can find a good bit of information about how to integrate a device, but I could not find much about accessing an existing device.  That doesn't mean it's not out there, just not able to find it so far.
